src/test/scala/nl.knaw.dans.easy.bag2deposit/ddm/ProvenanceV2Spec.scala

```diff
@@ -0,0 +1,112 @@
+  private val schema = {
+    val xsd: String = "target/dans-schema-lib/bag/metadata/prov/v2/provenance.xsd"
+    SchemaFactory
+      .newInstance(XMLConstants.W3C_XML_SCHEMA_NS_URI)
+      .newSchema(new StreamSource(new File(xsd)))
+  }
+
+  // FixedCurrentDateTimeSupport is not effective for a val
+  private def provenanceBuilder = Provenance("EasyConvertBagToDepositApp", "1.0.5")
+
+  "Provenance" should "change namespace" in {
+    val ddmIn = {
+      <ddm:DDM xmlns:ddm="http://easy.dans.knaw.nl/schemas/md/ddm/"
+               xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+               xsi:schemaLocation="http://easy.dans.knaw.nl/schemas/md/ddm/ https://easy.dans.knaw.nl/schemas/md/ddm/ddm.xsd"
+               xmlns:gml="http://www.opengis.net/gml"
+      >
+        <ddm:profile>
+          <dc:title>blabla</dc:title>
+        </ddm:profile>
+      </ddm:DDM>
+    }
+
+    val expectedDdm = <ddm:DDM xsi:schemaLocation={ s"$ddmV2namespace $ddmV2Location"}
+                               xmlns:ddm={ ddmV2namespace }
+                               xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+                               xmlns:gml="http://www.opengis.net/gml"
+    >
+      <ddm:profile xmlns:gml="http://www.opengis.net/gml"
+                   xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+                   xmlns:ddm="http://easy.dans.knaw.nl/schemas/md/ddm/">
+        <dc:title>blabla</dc:title>
+        <ddm:personalData present="Yes"/>
+      </ddm:profile>
+    </ddm:DDM>
+
+    val expectedProv = {
+      <prov:provenance xsi:schemaLocation="http://schemas.dans.knaw.nl/bag/metadata/prov-v2/ http://schemas.dans.knaw.nl/bag/metadata/prov/v2/provenance.xsd"
+                       xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+                       xmlns:prov="http://schemas.dans.knaw.nl/bag/metadata/prov-v2/"
+      >
+        <prov:migration app="EasyConvertBagToDepositApp" version="1.0.5" date="2020-02-02">
+        <prov:file scheme={ ddmV2namespace } xmlns=""
+                   xmlns:ddm={ ddmV2namespace }
+                   xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+                   xmlns:gml="http://www.opengis.net/gml"
+        >
+        <prov:old/>
+        <prov:new><ddm:personalData present="Yes"/></prov:new>
+          </prov:file>
+        </prov:migration>
+      </prov:provenance>
+    }
+
+    // a few steps of EasyConvertBagToDepositApp.addProps
+    val ddmOut = testConfig("archaeology").copy(ddmVersion = V2).ddmTransformer
+      .transform(ddmIn, "easy-dataset:123", containsPrivacySensitiveData = "true")
+      .getOrElse(fail("no DDM returned"))
+    val actualProv = provenanceBuilder.collectChangesInXmls(List(Provenance.compareDDM(ddmIn, ddmOut)))
+
+    logger.debug(ddmOut.serialize)
+    logger.debug(actualProv.serialize)
```

Comment:
Each element has namespace declarations. They are only desired at the root element or perhaps at the `<prov:file>`. Not sure whether I solved this before and lost it now.